### PR TITLE
allow git commit to succeed if there are no...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

allow git commit to succeed if there are no changes to commit (or really, pass in all cases)

Change-Id: I7d48ef9cc4ced12a47ddde47a3d806b599457dbc
Signed-off-by: nickboldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit messasge above for issue number)

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by nickboldt** [comment]
Not sure if this is a safe change to make, so wanted to discuss.

Problem I'm seeing is that if I try to re-release che operator 7.22.2, I get:

```
+ echo '[INFO] List of changed files:'
[INFO] List of changed files:
+ git status -s
+ echo '[INFO] Commiting changes'
[INFO] Commiting changes
+ git commit -am 'Update defaults tags to 7.22.2' --signoff
On branch 7.22.2-release
nothing to commit, working tree clean
Error: Process completed with exit code 1.
```

and the action fails in https://github.com/eclipse/che-operator/runs/1499748982?check_suite_focus=true

File: make-release.sh [hunk 1]
```diff
@@ -155,7 +155,7 @@ releaseOperatorCode() {
   git status -s
 
   echo "[INFO] Commiting changes"
-  git commit -am "Update defaults tags to "$RELEASE --signoff
+  git commit -am "Update defaults tags to "$RELEASE --signoff || true
```
Comment (benoitf) [inline make-release.sh:158]: @nboldt 
what about
```bash
if [[ `git status --porcelain` ]]; then
  echo "[INFO] Commiting changes"
  git commit -am "Update defaults tags to "$RELEASE --signoff	  git commit -am "Update defaults tags to "$RELEASE --signoff 
fi
```

Comment (nickboldt) [reply make-release.sh:158]: good idea.

**Review by tolusha — APPROVED** [review]

**Pull request merged**